crypto/fipsmodule/service_indicator/service_indicator.c

```diff
@@ -0,0 +1,85 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <openssl/crypto.h>
+#include <openssl/service_indicator.h>
+
+#if defined(AWSLC_FIPS)
+int awslc_fips_service_indicator_get_counter(void) {
+  struct fips_service_indicator_state *indicator =
+      CRYPTO_get_thread_local(AWSLC_THREAD_LOCAL_FIPS_SERVICE_INDICATOR_STATE);
+  if (!indicator) {
+    if(!awslc_fips_service_indicator_init_state()) {
+      return 0;
+    }
+    indicator = CRYPTO_get_thread_local(AWSLC_THREAD_LOCAL_FIPS_SERVICE_INDICATOR_STATE);
+  }
+  return indicator->counter;
+}
+
+int awslc_fips_service_indicator_get_serviceID(void) {
+  struct fips_service_indicator_state *indicator =
+      CRYPTO_get_thread_local(AWSLC_THREAD_LOCAL_FIPS_SERVICE_INDICATOR_STATE);
+  if (!indicator) {
+    if(!awslc_fips_service_indicator_init_state()) {
+      return 0;
+    }
+    indicator = CRYPTO_get_thread_local(AWSLC_THREAD_LOCAL_FIPS_SERVICE_INDICATOR_STATE);
+  }
+  return indicator->serviceID;
+}
+
+int awslc_fips_service_indicator_reset_state(void) {
```

Comment:
you can make this function static since it's not supposed to be used outside of the file.